frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -585,6 +591,11 @@ global.user_groups.add(backend);
         expected_value = [hamletcharacters, backend];
         assert.deepEqual(actual_value, expected_value);
 
+        fake_this = { completing: 'mention', token: 'ham' };
+        actual_value = options.sorter.call(fake_this, [hamlet, hamletcharacters]);
+        expected_value = [hamletcharacters, hamlet];
+        assert.deepEqual(actual_value, expected_value);
```

Comment:
I'd like this test to have users that are more like 

"Mr. Hamburger <mr.hamburger@zulip.com>", and "Sham User <sham@zulip.com>", where the `ham` appears later in the name, because those are cases that we might want to change in future versions...

Reply:
Yes, and I should also add "zeus@hamel.com" to test the case mentioned in the issue. I'll update the PR after the feedback from @showell

Reply:
You might as well do this now; he'll want to see the tests for review.

Reply:
Yeah, for the typeahead stuff, paying extra attention to the node tests can be really helpful for future maintenance.  And I agree with Tim's suggestions on having things like `ham` later in the name, including variations like `ham` is in the name but not the email, and vice versa.  There are basically 9 combos of "ha" being at-the-front-of/in-the-middle-of/not-in the name/email.  You don't have to test all 9 combos, but you should cover enough interesting cases to have confidence here.